### PR TITLE
Fix build issues with no-dh, no-dsa and no-ec

### PR DESCRIPTION
**Pull request by levitte**
##### Checklist

- [x] CLA is signed

### CODE REVIEW

File: fuzz/server.c [hunk 1]
```diff
@@ -507,8 +517,12 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
     const uint8_t *bufp;
     EVP_PKEY *pkey;
     X509 *cert;
+#ifndef OPENSSL_NO_DSA
```
Comment (kroeckx) [inline fuzz/server.c]: I think the NO_DSA doesn't make sense at this line (and some others), I don't think the EC(DSA) code shares anything with DSA code.

Comment (levitte) [inline fuzz/server.c]: You're right.  Fixing

**Comment by levitte** [comment]
Please have a look now, @kroeckx

**Review by kroeckx — APPROVED** [review]

**Comment by levitte** [comment]
Merged

**Pull request closed without merging**